modules/install/join_k3s_agent.sh

```diff
@@ -4,26 +4,56 @@
 #set -x
 # PS4='+(${LINENO}): '
 
+set -x
+echo "$@"
+
+node_os=${1}
+server_ip=${2}
+token=${3}
+public_ip=${4}
+private_ip=${5}
+ipv6_ip=${6}
+install_mode=${7}
+version=${8}
+channel=${9}
+worker_flags=${10}
+rhel_username=${12}
+rhel_password=${13}
+
 mkdir -p /etc/rancher/k3s
 
 create_config() {
-  local server_ip="${1}"
-  local token="${2}"
-
+  hostname=$(hostname -f)
   cat <<EOF >>/etc/rancher/k3s/config.yaml
-server: https://${1}:6443
-token:  "${2}"
+server: "https://$server_ip:6443"
+token:  "$token"
+node-name: $hostname
 EOF
 }
 
 add_config() {
-  local worker_flags="${1}"
-
   if [[ -n "$worker_flags" ]] && [[ "$worker_flags" == *":"* ]]
+  then
+    echo "$worker_flags"
```

Comment:
Can probably remove this line -- we don't need worker flags to be output here since we cat the entire config.yaml anyway

Reply:
Updated